File: backoffice/src/shared/infrastructure/eventBus/RabbitmqQueueNameFormatter.py
"""
 *
 * Libraries 
 *
"""

from ..events.DomainEventInformation import DomainEventInformation

"""
 *
 * Class
 *
"""

class RabbitmqQueueNameFormatter:

    """
     *
     * Methods 
     *
    """

    def format( self, eventInformation : DomainEventInformation ) -> str:
        return eventInformation.formatRabbitmqQueueName()

    def formatDeadLetter( self, eventInformation : DomainEventInformation ) -> str:
        return 'dead.letter.{}'.format( self.format( eventInformation ) )

File: preparation/src/shared/infrastructure/eventBus/RabbitmqQueueNameFormatter.cs
/* 
 * 
 * Class
 *
*/

namespace preparation.src.shared.infrastructure {

    public class RabbitmqQueueNameFormatter { 

        /* 
         * 
         * Methods
         *
        */

        public string format( DomainEventInformation eventInformation ) {
            return eventInformation.formatRabbitmqQueueName();
        }

        public string formatDeadLetter( DomainEventInformation eventInformation ) {
            return string.Format( "dead.letter.{0}", format( eventInformation ) );
        }
    
    }

}